packages/amplify-codegen/src/commands/generateStatementsAndType.js

```diff
@@ -47,6 +47,10 @@ async function generateStatementsAndTypes(context, forceDownloadSchema, maxDepth
     ({ projectPath } = context.amplify.getEnvInfo());
   }
 
+  await context.amplify.executeProviderUtils(context, 'awscloudformation', 'compileSchema', {
+    forceCompile: true,
+  });
+
```

Comment:
We might need to add a check here to ensure that there is an valid GraphQL API in the project before executing this. The CLI allows customers to add external api using `amplify add codegen --apiId <your-api-id>`. If the API is an external API, then we download the introspection schema and generate types and statements.